pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -346,15 +438,23 @@ func newCPUAccumulator() *cpuAccumulator {
 
 // AddCores adds logical cores from the slice of *cpu.ProcessorCore to a CPUset till the cpuset size is equal to the max value specified
 // In case the max is specified as allCores, all the cores from the slice of *cpu.ProcessorCore are added to the CPUSet
-func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) {
+func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) int {
+	return ca.AddCoresWithFilter(max, cores, noFilerLogicalProcessors)
+}
+
+func (ca *cpuAccumulator) AddCoresWithFilter(max int, cores []*cpu.ProcessorCore, filterLogicalProcessor func(int, int) bool) int {
+	initialCount := ca.count
 	for _, processorCore := range cores {
-		for _, core := range processorCore.LogicalProcessors {
+		for index, logicalProcessorId := range processorCore.LogicalProcessors {
 			if ca.count < max || max == allCores {
-				ca.builder.Add(core)
-				ca.count++
+				if filterLogicalProcessor(index, logicalProcessorId) {
+					ca.builder.Add(logicalProcessorId)
```

Comment:
I think there could be an error here with cpuAccumulator.
If `logicalProcessorId` is already present in the builder it will be updated but the number of elements will not be increased therefore `ca.count` will have a wrong value.

If `build.Add` could return the number of elements added this would help ... otherwise I think ca.count should be calculated as `builder.Size` to avoid the problem.
@fromanirh, @Tal-or @swatisehgal  what do you think?

Reply:
As this is an offline tool and I do not think we expect cpuAccumulator to have thousands of elements, could we go with using ca.builder.Size() to update ca.count in the meantime?

Reply:
I guess we can. I was (and still am) thinking more about code clarity (`cpuset.Builder` reuse is effectively doing more harm than good now) rather than performance.

Reply:
Make a change proposal here : https://github.com/openshift/cluster-node-tuning-operator/pull/354/commits/954adafefd73bcae947dcaaeea1c745d6a82687e